### PR TITLE
[fix](storage vault) Fix missing use_path_style when create storage vault

### PR DESCRIPTION
**Pull request by gavinchou**
### What problem does this PR solve?

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [x] Add branch pick label

### CODE REVIEW

**Comment by gavinchou** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/S3Properties.java [hunk 1]
```diff
@@ -343,6 +343,9 @@ public static Cloud.ObjectStoreInfoPB.Builder getObjStoreInfoPB(Map<String, Stri
                     || value.equalsIgnoreCase("false"),
                     "Invalid use_path_style value: %s only 'true' or 'false' is acceptable", value);
             builder.setUsePathStyle(value.equalsIgnoreCase("true"));
+        } else {
+            // use path style
+            builder.setUsePathStyle(true);
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/S3Properties.java:348]:
```suggestion
```

**Comment by gavinchou** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by luwei16 — APPROVED** [review]

**Pull request merged**